pkg/dockerregistry/server/configuration/configuration_test.go

```diff
@@ -0,0 +1,90 @@
+package configuration
+
+import (
+	"bytes"
+	"os"
+	"testing"
+)
+
+var configYamlV0_1 = `
+version: 0.1
+http:
+  addr: :5000
+  relativeurls: true
+storage:
+  inmemory: {}
+openshift:
+  version: 1.0
+  metrics:
+    enabled: true
+    secret: TopSecretToken
+`
+
+func TestConfigurationParser(t *testing.T) {
+	configFile := bytes.NewBufferString(configYamlV0_1)
+
+	dockerConfig, extraConfig, err := Parse(configFile)
+	if err != nil {
+		t.Fatalf("unexpected error parsing configuration file: %s", err)
+	}
+
+	if !dockerConfig.HTTP.RelativeURLs {
+		t.Fatalf("unexpected value: dockerConfig.HTTP.RelativeURLs != true")
+	}
+
+	if !extraConfig.Metrics.Enabled {
+		t.Fatalf("unexpected value: extraConfig.Metrics.Enabled != true")
+	}
+}
+
+func TestConfigurationOverwriteEnv(t *testing.T) {
+	os.Setenv("REGISTRY_OPENSHIFT_METRICS_ENABLED", "false")
+	defer os.Unsetenv("REGISTRY_OPENSHIFT_METRICS_ENABLED")
+
+	configFile := bytes.NewBufferString(configYamlV0_1)
+
+	_, extraConfig, err := Parse(configFile)
+	if err != nil {
+		t.Fatalf("unexpected error parsing configuration file: %s", err)
+	}
+	if extraConfig.Metrics.Enabled {
+		t.Fatalf("unexpected value: extraConfig.Metrics.Enabled != false")
+	}
+}
+
+func TestDockerConfigurationError(t *testing.T) {
+	var badDockerConfigYamlV0_1 = `
+version: 0.1
+http:
+  addr: :5000
+  relativeurls: "true"
+storage:
+  inmemory: {}
```

Comment:
Let's not make the `openshift` section mandatory for now.

I know...I said no more configuration comments, but I can't resist.